tests/components/gree/test_climate.py

```diff
@@ -843,6 +870,40 @@ async def test_update_swing_mode(
     assert state.attributes.get(ATTR_SWING_MODE) == swing_mode
 
 
+async def test_coordinator_update_handler(
+    hass: HomeAssistant, discovery, device
+) -> None:
+    """Test for coordinator update handler."""
+    await async_setup_gree(hass)
+    await hass.async_block_till_done()
+
+    entity: GreeClimateEntity = hass.data[DOMAIN].get_entity(ENTITY_ID)
+    assert entity is not None
+
+    # Initial state
+    assert entity.temperature_unit == UnitOfTemperature.CELSIUS
+    assert entity.min_temp == TEMP_MIN
+    assert entity.max_temp == TEMP_MAX
+
+    # Set unit to FAHRENHEIT
+    device().temperature_units = 1
+    entity.coordinator.async_set_updated_data(UnitOfTemperature.FAHRENHEIT)
```

Comment:
Don't interact with the coordinator either. Move time forward or patch and call the library callback to refresh the coordinator.